plugin/db/pg/pg.go

```diff
@@ -438,19 +438,22 @@ func (driver *Driver) NeedsSetupMigration(ctx context.Context) (bool, error) {
 	return util.NeedsSetupMigrationSchema(ctx, driver.db, query)
 }
 
-func (driver *Driver) hasBytebaseDatabase(ctx context.Context) (bool, error) {
+// hasDatabase finds if there is a database named `dbName` in this instance
+func (driver *Driver) hasDatabase(ctx context.Context, dbName string) (bool, error) {
```

Comment:
I would stick with a single function since it's only called by hasBytebaseDatabase

Extracting a function with a single caller has the overhead of letting users to check who are the callers.

Reply:
I think the function in `plugin/db` should be generic enough. If `hasBytebaseDatabase()` in some special 
 purpose file like `pg_engine.go`, I will find it resonable.

Reply:
More, I think the caller should call `hasDatabase("bytebase")` instead of `hasBytebaseDatabase()`.

Reply:
If we need to call hasDatabase("xxx") , hasDatabase("yyy") , then it will make more sense.

However, currently, we only need a single hasBytebaseDatabase, thus generalizing it doesn't bring extra benefit, while just adding another layer of indirection.

Reply:
SGTM.